torch/_functorch/aot_autograd.py

```diff
@@ -624,9 +637,15 @@ def __post_init__(self):
         self.num_outputs_aliased = (
             self.num_outputs_aliased_to_inputs + self.num_outputs_aliased_to_intermediates
         )
-        self.num_mutated_data_inputs = len(
-            [x for x in self.input_info if x.mutates_data]
-        )
+        if not self.is_subclass and self.keep_input_mutations and self.is_train:
```

Comment:
A comment here would be nice describing the situation:

In fact, it would be great to have a larger comment here laying out exactly when input mutations are / are not in the graph in the first place (with "training graph where input mutation happens to a subclass" being one of those cases0.